earthpy/clip.py

```diff
@@ -74,6 +110,65 @@ def _clip_line_poly(shp, clip_obj):
     return clipped[clipped.geometry.notnull()]
 
 
+def _clip_multi_poly_line(shp, clip_obj):
```

Comment:
I agree the implementation seems a wee bit hacky. Can it be simplified by relying on the concatenation of zero row data frames instead of conditional logic? e.g., 

```python
def _clip_multi_poly_line(shp, clip_obj):
    """..."""
    # Clip multi polygons
    clipped = _clip_line_poly(shp.explode().reset_index(level=[1]), clip_obj)

    lines = clipped[
        (clipped.geometry.type == "MultiLineString")
        | (clipped.geometry.type == "LineString")
    ]
    line_diss = lines.dissolve(by=[lines.index]).drop(columns="level_1")

    polys = clipped[clipped.geometry.type == "Polygon"]
    poly_diss = polys.dissolve(by=[polys.index]).drop(columns="level_1")

    return gpd.GeoDataFrame(
        pd.concat([poly_diss, line_diss], ignore_index=True)
    )
```

Note that I tested this locally and all tests pass. If this simpler function is incorrect in some way, we should add some tests to capture that.

Reply:
i'll test this. we have the edge case in our tests where a sliver (a line) and a poly are returned. Let me see if it works when we just have a poly.  although i did think i made a poly only specific test...

Reply:
gosh... so i didn't realize that zero-row dataframe concatenation would work @mbjoseph but it does!! thank you. this code is so much cleaner. i've implemented this. i have a few other questions for you WRT to fixtures that i'll add in a moment